src/components/v5/common/ActionSidebar/partials/forms/ManageTokensForm/partials/TokensTable/TokensTable.tsx

```diff
@@ -0,0 +1,133 @@
+import { Plus, Trash, X } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import React, { useEffect, type FC } from 'react';
+import { useFieldArray, useFormContext, useWatch } from 'react-hook-form';
+
+import { useTablet } from '~hooks/index.ts';
+import { formatText } from '~utils/intl.ts';
+import Table from '~v5/common/Table/index.ts';
+import { TokenStatus } from '~v5/common/types.ts';
+import Button from '~v5/shared/Button/index.ts';
+
+import { useTokensTableColumns } from './hooks.tsx';
+import { type TokensTableModel, type TokensTableProps } from './types.ts';
+
+const displayName = 'v5.common.ActionsContent.partials.TokensTable';
+
+const TokensTable: FC<TokensTableProps> = ({
+  name,
+  shouldShowMenu = () => true,
+  isDisabled,
+}) => {
+  const isTablet = useTablet();
+  const fieldArrayMethods = useFieldArray({
+    name,
+  });
+  const data: TokensTableModel[] = fieldArrayMethods.fields.map(({ id }) => ({
+    key: id,
+  }));
+  const { getFieldState, watch, trigger } = useFormContext();
+  const fieldState = getFieldState(name);
+  const value = useWatch({ name }) || [];
+  const columns = useTokensTableColumns(name, value);
+  const getMenuProps = ({ index }) => {
+    const shouldShow = shouldShowMenu(value[index]?.status);
+
+    const editPreviouslyAddedToken = [
+      TokenStatus.Unaffected,
+      TokenStatus.Removed,
+    ].includes(value[index]?.status);
+
+    const hasRemovedLabel = value[index]?.status === TokenStatus.Removed;
+
+    return shouldShow && !isDisabled
+      ? {
+          cardClassName: 'sm:min-w-[9.625rem]',
+          items: [
+            {
+              key: 'remove',
+              onClick: () =>
+                editPreviouslyAddedToken
+                  ? fieldArrayMethods.update(index, {
+                      ...value[index],
+                      status: hasRemovedLabel
+                        ? TokenStatus.Unaffected
+                        : TokenStatus.Removed,
+                    })
+                  : fieldArrayMethods.remove(index),
+              label: formatText({
+                id: hasRemovedLabel ? 'table.row.cancel' : 'table.row.remove',
+              }),
+              icon: hasRemovedLabel ? X : Trash,
+            },
+          ],
+        }
+      : undefined;
+  };
+
+  // Added to trigger validation on token change to check for duplicates in previous fields
+  useEffect(() => {
+    const subscription = watch((formValues, { name: fieldName = '' }) => {
+      const fieldNameParts = fieldName.split('.');
+
+      if (fieldNameParts[0] !== name) {
+        return;
+      }
+
+      formValues.selectedTokenAddresses.forEach(({ token }, index) => {
+        if (!token) {
+          return;
+        }
+
+        trigger(`${name}.${index}.token`);
+      });
+    });
+
+    return () => subscription.unsubscribe();
+  }, [name, trigger, watch]);
```

Comment:
This component has both `useWatch` and the callback version of watch API, is there a reason for it?

The [docs](https://react-hook-form.com/docs/usewatch) suggest `useWatch` can offer better performance, and I'd lean towards it for better readability/consistency.